broker/src/broker_server.cpp

```diff
@@ -0,0 +1,525 @@
+                auto beerocks_magic = *(uint32_t *)(tlv_vendor_specific.payload());
+                swap_32(beerocks_magic);
+                if (beerocks_magic != message::MESSAGE_MAGIC) {
+                    LOG(WARNING) << "mismatch magic " << std::hex << int(beerocks_magic)
+                                 << " != " << int(message::MESSAGE_MAGIC) << std::dec;
+                    ret = false;
+                    break;
+                }
+
+            } else {
+                LOG(INFO) << "Not an Intel vendor specific message!";
+            }
+
+            // cancel the swap we did
+            tlv_vendor_specific.class_swap();
+        } else if (static_cast<ieee1905_1::eTlvType>(type) ==
+                       ieee1905_1::eTlvType::TLV_END_OF_MESSAGE &&
+                   length == 0) {
+            return true;
+        }
+
+        // set the next tlv
+        tlv  = (ieee1905_1::sTlvHeader *)((uint8_t *)tlv + sizeof(ieee1905_1::sTlvHeader) + length);
+        type = tlv->type;
+        length = tlv->length;
+
+    } while ((uint8_t *)tlv <= (uint8_t *)first_tlv + uds_header->length);
+
+    LOG(ERROR) << "TLV end of message not found! tlv_type=" << int(tlv->type)
+               << ", tlv_length=" << int(tlv->length); // << ", " << print_cmdu_types(uds_header);
+
+    std::ptrdiff_t available_bytes = uds_header->length + sizeof(message::sUdsHeader);
+    LOG(DEBUG) << "hex_dump (" + std::to_string(available_bytes) + " bytes):" << std::endl
+               << utils::dump_buffer(static_cast<const uint8_t *>(cmdu), available_bytes);
+
+    return ret;
+}
+
+static bool is_restricted_type(uint32_t type)
+{
+    // TODO: Until VS message filtering is implemented (based on action and action_op)
+    //       assume there are no restricted types. After the implementation,
+    //       subscribing to BROKER specific message should be restricted.
+
+    return false;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+/////////////////////////// Broker Implementation ////////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+broker_server_thread::broker_server_thread(std::string broker_uds_path)
+    : m_broker_uds_path(broker_uds_path)
+{
+}
+
+broker_server_thread::~broker_server_thread()
+{
+    // Stop the thread
+    stop();
+}
+
+bool broker_server_thread::init()
+{
+    // Connect to the message broker
+    if (!(m_broker_socket_uds =
+              std::make_shared<SocketServer>(m_broker_uds_path, MAX_SERVER_CONNECTIONS))) {
+        LOG(FATAL) << "Failed allocating memory!";
+        return false;
+    }
+
+    // Check for connection errors
+    auto error_msg = m_broker_socket_uds->getError();
+    if (!error_msg.empty()) {
+        LOG(ERROR) << "Failed connecting to the broker using UDS: " << m_broker_uds_path
+                   << " [ERROR: " << error_msg << "]";
```

Comment:
Again, this is pretty fatal, so
```suggestion
    LOG_IF (FATAL, !error_msg.empty())  << "Failed connecting to the broker using UDS: " << m_broker_uds_path
                   << " [ERROR: " << error_msg << "]";
```

Reply:
True, changed.